### PR TITLE
Add support for point-in-time search in OpenSearch connector

### LINKED ISSUE #20834: Can we use PIT search instead of scroll search in opensearch connector?
Can we use PIT search instead of scroll search in opensearch connector?
It is mentioned that PIT search is preferred when deep pagination is done:
`Point in Time (PIT) with search_after is the preferred pagination method in OpenSearch, especially for deep pagination. It bypasses the limitations of all other methods because it operates on a dataset that is frozen in time, it is not bound to a query, and it supports consistent pagination going forward and backward.`
[https://opensearch.org/docs/latest/search-plugins/searching-data/paginate/](https://opensearch.org/docs/latest/search-plugins/searching-data/paginate/)

It is mentioned to use PIT in elastic discussion also: https://discuss.elastic.co/t/sliced-scroll-with-sort/284874
It would be nice to have a feature to enable search using PIT to gain performance benefits when searching on huge datasets.

### PR DESCRIPTION
**Pull request by aravind-musigumpula**
## Description

Fixes #20834

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/OpenSearchPageSourceProvider.java [hunk 1]
```diff
@@ -66,14 +70,30 @@ public ConnectorPageSource createPageSource(
         if (columns.isEmpty()) {
             return new CountQueryPageSource(client, opensearchTable, opensearchSplit);
         }
+        System.out.println("search Type: " + config.getSearchType());
```
Comment (Praveen2112) [inline plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/OpenSearchPageSourceProvider.java:73]: Please remove this line.

File: plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/OpenSearchPageSourceProvider.java [hunk 2]
```diff
@@ -37,11 +38,14 @@ public class OpenSearchPageSourceProvider
     private final OpenSearchClient client;
     private final TypeManager typeManager;
 
+    private final OpenSearchConfig config;
```
Comment (Praveen2112) [inline plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/OpenSearchPageSourceProvider.java:41]: Avoid maintaining config object as a field - as they are mutable in the future.

File: plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/OpenSearchSplitManager.java [hunk 3]
```diff
@@ -52,11 +57,37 @@ public ConnectorSplitSource getSplits(
...
 
         if (tableHandle.getType().equals(OpenSearchTableHandle.Type.QUERY)) {
-            return new FixedSplitSource(new OpenSearchSplit(tableHandle.getIndex(), 0, Optional.empty()));
+            return new FixedSplitSource(new OpenSearchSplit(tableHandle.getIndex(), 0, Optional.empty(), 0, null));
         }
-        List<OpenSearchSplit> splits = client.getSearchShards(tableHandle.getIndex()).stream()
-                .map(shard -> new OpenSearchSplit(shard.getIndex(), shard.getId(), shard.getAddress()))
-                .collect(toImmutableList());
+
+        System.out.println("config in split manager: " + config.toString());
```
Comment (Praveen2112) [inline plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/OpenSearchSplitManager.java:63]: Please remove this line.

**Review by Praveen2112 — COMMENTED** [review]
Just skimmed it. Please remove `System.out.println` from the code base

**Comment by Praveen2112** [comment]
Thanks a lot for working on this.

**Comment by aravind-musigumpula** [comment]
Will raise a new PR, with more changes. Kindly ignore this PR @Praveen2112

**Comment by mosabua** [comment]
Feel free to close this PR and replace it with another @aravind-musigumpula .. just keep in mind that smaller PRs are easier to review and therefore probably get merged faster.

**Comment by aravind-musigumpula** [comment]
Will reopen this with another PR

**Pull request closed without merging**